packages/table/src/cell/cell.tsx

```diff
@@ -53,17 +53,35 @@ export interface ICellProps extends IIntentProps, IProps {
     wrapText?: boolean;
 }
 
+// don't include "style" in here because it can't be shallowly compared
+const UPDATE_PROPS_KEYS = [
+    "className",
+    "intent",
+    "interactive",
+    "loading",
+    "tooltip",
+    "truncated",
+    "wrapText"
+];
+
 export type ICellRenderer = (rowIndex: number, columnIndex: number) => React.ReactElement<ICellProps>;
 
 export const emptyCellRenderer = () => <Cell />;
 
-@PureRender
 export class Cell extends React.Component<ICellProps, {}> {
     public static defaultProps = {
         truncated: true,
         wrapText: false,
     };
 
+    public shouldComponentUpdate(nextProps: ICellProps) {
+        // shallowly comparable props like "className" tend not to change in the default table
+        // implementation, so do that check last with hope that we return earlier and avoid it
```

Comment:
I believe this text is just copied-and-pasted from a similar series of checks we do in another component. Makes sense to do the shallow stuff first I guess. (Original motivation was to first check the stuff most likely to have changed.)